cms/static/cms/js/modules/cms.message.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright https://github.com/divio/django-cms
+ */
+
+// #############################################################################
+// NAMESPACES
+/**
+ * @module CMS
+ */
+var CMS = window.CMS || {};
+
+// #############################################################################
+// MODAL
+(function ($) {
+    'use strict';
+
+    // shorthand for jQuery(document).ready();
+    $(function () {
+        /**
+         * Displays a message underneath the toolbar
+         *
+         * @class Message
+         * @namespace CMS
+         */
+        CMS.Message = new CMS.Class({
+
+            options: {
+                messageDuration: 300,
+                messageDelay: 3000
+            },
+
+            initialize: function initialize(options) {
+                this.options = $.extend(true, {}, this.options, options);
+                this.config = CMS.config;
+                this.settings = CMS.settings;
+
+                // elements
+                this._setupUI();
+            },
+
+            /**
+             * Stores all jQuery references within `this.ui`.
+             *
+             * @method _setupUI
+             * @private
+             */
+            _setupUI: function _setupUI() {
+                var container = $('#cms-toolbar');
+                this.ui = {
+                    container: container,
+                    body: $('html'),
+                    toolbar: container.find('.cms-toolbar'),
+                    messages: container.find('.cms-messages')
+                };
+            },
+
+            /**
+             * Opens a message window underneath the toolbar.
```

Comment:
we should tell somewhere here that if delay is 0 it's actually not gonna hide until you explicitly close it